File: features/support/pages/imedidata/imedidata.rb
require_relative File.join(Rails.root, 'features', 'support', 'page.rb').to_s
require 'memoize'

module JanusHelper
  class Imedidata < Page
    # include Memoize
    def self.visit
      page.visit JanusNavigation::CONFIG_MAPPING['apps']['login']
      new
      # self.memoize(:mapper)
    end

    def login
      selector, config = mapper
      selector = selector['Sign In']

      fill_in selector['username']['label'], with: config['username']
      fill_in selector['password']['label'], with: config['password']
      click_button selector['submit']['text']
    end

    def logout
      selector, config = mapper
      selector = selector['Header']

      within selector['section']['panel'] do
        click_link config['username']
        page.find(selector['logout']['link'],visible: true).click
      end
    end

    private

    def mapper
      @selector ||= JanusNavigation::SELECTOR_MAPPING['Imedidata']
      @config ||= JanusNavigation::CONFIG_MAPPING['apps']
      return @selector, @config
    end
  end
end
World(JanusHelper)

File: features/support/pages/javascript/imedidata/imedidata.js
/**
 * Created by scalixte on 4/30/15.
 */
'use strict';

var information = information || require('../../../../support/javascript/information');
var support = support || require('../../../../support/javascript/support');

var Imedidata = function(){
    this.config = information.config['apps'];
    this.selector = information.selector['Imedidata'];
};

Imedidata.prototype.visit=function(that, callback){};

Imedidata.prototype.login=function(that, callback){
    try {
        var self = this;
        support.ignoreSynchronization(that, true, function (result) {});
        support.get(that, this.config['login'], function (result) {
            setTimeout(callback, 1000);
        });
        support.findByName(that,'session[username]', function(result){result.sendKeys(self.config['username']);});
        support.findByName(that,'session[password]', function(result){result.sendKeys(self.config['password']);});
        support.findByCSS(that,'#create_session_link', function(result){result.click();});
    } catch (e) {
        console.log(e);
    }
};

Imedidata.prototype.logout=function(that, callback){
    try{
        var self = this;
        var that = that;
        support.ignoreSynchronization(that, true, function (result) {});
        support.findByCSS(that, this.selector['Header']['signout']['link'], function (result) {
            result.click().then(function () {
                support.findByCSS(that, self.selector['Header']['logout']['link'], function (result) {
                    result.click().then(function(){support.exit(that)});
                });
            });

        });



    } catch (e) {
        console.log(e);
    }
};


module.exports = new Imedidata();